### PR TITLE
increase lower bounds for stdlib, vcsrepo

### LINKED ISSUE #326: Puppet-python version dependency update
## Affected Puppet, Ruby, OS and module versions/distributions

- Puppet: 6
- Ruby:
- Distribution: 
- Module version: 7.0.2

## How to reproduce (e.g Puppet code you use)
When pinning the puppet-python module version to the latest, 6.0.0, a dependency failure occurs as this module requires puppet-python to be < 6.0.0

## What behaviour did you expect instead
I'd be highly appreciated if this dependency could be bumped to at least <= 6.0.0

### PR DESCRIPTION
**Pull request by kenyon**
Closes #326.

### CODE REVIEW

**Comment by t-valette** [comment]
Hello @kenyon,

I'm not sure puppet/puppetboard is compatible with puppet/python >= 6 as we still find `virtualenv` and `manage_virtualenv_package` in the [init file](https://github.com/voxpupuli/puppet-puppetboard/blob/master/manifests/init.pp#L197).
If you look at the changelog of puppet/python, we can see that these parameters are no longer present:
* `Drop python::virtualenv defined resource` from [changelog](https://github.com/voxpupuli/puppet-python/blob/master/CHANGELOG.md#v600-2021-04-03)
* no more reference to virtual in the [init file 6.0.0+](https://github.com/voxpupuli/puppet-python/blob/v6.0.0/manifests/init.pp)
* explanation on why virtualenv is now deprecated in the [PR](https://github.com/voxpupuli/puppet-python/pull/596)

Also, I tested with your PR and that's what I'm getting (with `mod 'puppet/python', :latest` in our Puppetfile) :
```
Error: Could not retrieve catalog from remote server: Error 500 on SERVER: Server Error: Evaluation Error: Error while evaluating a Resource Statement, Class[Python]:
  has no parameter named 'virtualenv'
  has no parameter named 'manage_virtualenv_package' (file: /.../production/modules/puppetboard/manifests/init.pp, line: 188, column: 5) on node puppet...
```

Here, I see two solutions for that (I'm not a puppet expert, maybe I wrong but I try to add some ideas):
* dropping virtualenv parameters for systems having puppet/python >= 6.0.0
* use `python::pyvenv` described [here](https://github.com/voxpupuli/puppet-python/blob/v6.0.0/manifests/pyvenv.pp) which is already present in puppet/python >= 4.1.0

**Comment by kenyon** [comment]
Indeed, I did not look for such incompatibilities. I figured we could find and fix those in other pull requests. So if you could submit a PR that would be great. 👍

File: metadata.json [hunk 1]
```diff
@@ -31,21 +31,21 @@
...
     },
     {
       "name": "puppet/python",
-      "version_requirement": ">= 4.1.0 < 6.0.0"
+      "version_requirement": ">= 4.1.0 < 7.0.0"
     },
     {
       "name": "puppetlabs/vcsrepo",
-      "version_requirement": ">= 1.3.1 < 4.0.0"
+      "version_requirement": ">= 1.3.1 < 5.0.0"
```
Comment (ghoneycutt) [inline metadata.json:48]: The lower bound should also change to `4.0.0` as they dropped puppet 5, same with stdlib which should change to `7.0.0` and python should be `6.0.0`

**Comment by kenyon** [comment]
Don't know why travis times out trying to download the puppetdb module. Maybe it needs https://github.com/puppetlabs/puppetlabs-puppetdb/pull/327

**Review by smortex — APPROVED** [review]

File: metadata.json [hunk 2]
```diff
@@ -31,21 +31,21 @@
...
       "name": "puppet",
-      "version_requirement": ">= 5.5.8 < 7.0.0"
+      "version_requirement": ">= 6.1.0 < 8.0.0"
     }
   ],
   "dependencies": [
     {
       "name": "puppetlabs/stdlib",
-      "version_requirement": ">= 4.13.1 < 7.0.0"
+      "version_requirement": ">= 7.0.0 < 8.0.0"
```
Comment (smortex) [inline metadata.json:40]: All the changes from these PR seems to have been in #317 except this lower bound.

File: metadata.json [hunk 3]
```diff
@@ -31,21 +31,21 @@
...
     },
     {
       "name": "puppet/python",
-      "version_requirement": ">= 4.1.0 < 6.0.0"
+      "version_requirement": ">= 6.0.0 < 7.0.0"
     },
     {
       "name": "puppetlabs/vcsrepo",
-      "version_requirement": ">= 1.3.1 < 4.0.0"
+      "version_requirement": ">= 4.0.0 < 5.0.0"
```
Comment (smortex) [inline metadata.json:48]: and this lower bound.

**Review by smortex — COMMENTED** [review]
After review of the differences which where part of #317, it seems two lower bounds where not included and if applicable, we can update this PR to fix this.

**Review by bastelfreak — APPROVED** [review]

**Pull request merged**